library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.kt

```diff
@@ -0,0 +1,81 @@
+package com.getstream.sdk.chat.adapter
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.view.MessageListView
+import com.getstream.sdk.chat.view.MessageListView.AttachmentClickListener
+import com.getstream.sdk.chat.view.MessageListView.BubbleHelper
+import com.getstream.sdk.chat.view.MessageListView.GiphySendListener
+import com.getstream.sdk.chat.view.MessageListView.MessageClickListener
+import com.getstream.sdk.chat.view.MessageListView.MessageLongClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReactionViewClickListener
+import com.getstream.sdk.chat.view.MessageListView.ReadStateClickListener
+import com.getstream.sdk.chat.view.MessageListViewStyle
+import io.getstream.chat.android.client.models.Channel
+
+class MessageListItemAdapter @JvmOverloads constructor(
+    private var context: Context,
+    var channel: Channel? = null,
+    private var messageListItemList: List<MessageListItem> = emptyList(),
+    var viewHolderFactory: MessageViewHolderFactory = MessageViewHolderFactory()
+) : RecyclerView.Adapter<BaseMessageListItemViewHolder<*>>() {
+
+    var bubbleHelper: BubbleHelper? = null
+
+    var messageClickListener: MessageClickListener? = null
+        set(value) {
+            if (style?.isReactionEnabled == true)
+                field = value
+        }
+    var messageLongClickListener: MessageLongClickListener? = null
```

Comment:
I were talking about those listeners in other PR, I think they shouldn't be into our Adapter, it doesn't need them.
On the other hand, would be great if we remove `null` value here, At the end those listener should be easy to convert into an empty function instead of a null value.
If they are mandatory and couldn't be added to our constructor because some reason, we can implement them as a function that throw an exception if no implementation is assigned.
At the end, no null values does our code is more readable

Reply:
Will refactor in CAS-237, where these will be moved from the adapter to the factory altogether